spec/rspec/core/bisect/coordinator_spec.rb

```diff
@@ -75,32 +78,82 @@ def find_minimal_repro(output, formatter=Formatters::BisectProgressFormatter)
+        |Round 3: bisecting over non-failing examples 1-2
+        | - Running: rspec 2.rb[1:1] 3.rb[1:1] 4.rb[1:1] 5.rb[1:1] (n.nnnn seconds)
+        | - Running: rspec 1.rb[1:1] 2.rb[1:1] 4.rb[1:1] 5.rb[1:1] (n.nnnn seconds)
+        | - Examples we can safely ignore (1):
         |    - 3.rb[1:1]
-        | - Remaining non-failing examples (1):
+        | - Remaining non-failing examples (2):
+        |    - 1.rb[1:1]
         |    - 4.rb[1:1]
-        | - Round finished (n.nnnn seconds)
-        |Round 3: searching for 1 non-failing example (of 1) to ignore:
-        | - Running: rspec 2.rb[1:1] 5.rb[1:1] (n.nnnn seconds)
-        | - Round finished (n.nnnn seconds)
-        |Bisect complete! Reduced necessary non-failing examples from 6 to 1 in n.nnnn seconds.
+        |Bisect complete! Reduced necessary non-failing examples from 6 to 2 in n.nnnn seconds.
         |
         |The minimal reproduction command is:
-        |  rspec 2.rb[1:1] 4.rb[1:1] 5.rb[1:1]
+        |  rspec 1.rb[1:1] 2.rb[1:1] 4.rb[1:1] 5.rb[1:1]
       EOS
     end
 
+    context "with an order-independent failure" do
+      it "detects the independent case and prints the minimal reproduction" do
+        fake_runner.dependent_failures = {}
+        output = StringIO.new
+        find_minimal_repro(output)
+        output = normalize_durations(output.string)
+
+        expect(output).to eq(<<-EOS.gsub(/^\s+\|/, ''))
+        |Bisect started using options: ""
+        |Running suite to find failures... (n.nnnn seconds)
+        |Starting bisect with 1 failing example and 7 non-failing examples.
+        |Checking that failure(s) are order-dependent... failure is not order-dependent
+        |
+        |Bisect complete! Reduced necessary non-failing examples from 7 to 0 in n.nnnn seconds.
+        |
+        |The minimal reproduction command is:
+        |  rspec 2.rb[1:1]
+        EOS
+      end
+
+      it "can use the debug formatter for detailed output" do
+        fake_runner.dependent_failures = {}
+        output = StringIO.new
+        find_minimal_repro(output, Formatters::BisectDebugFormatter)
+        output = normalize_durations(output.string)
+
+        expect(output).to eq(<<-EOS.gsub(/^\s+\|/, ''))
+        |Bisect started using options: ""
+        |Running suite to find failures... (n.nnnn seconds)
+        | - Failing examples (1):
+        |    - 2.rb[1:1]
+        | - Non-failing examples (7):
+        |    - 1.rb[1:1]
+        |    - 3.rb[1:1]
+        |    - 4.rb[1:1]
+        |    - 5.rb[1:1]
+        |    - 6.rb[1:1]
+        |    - 7.rb[1:1]
+        |    - 8.rb[1:1]
+        |Checking that failure(s) are order-dependent..
+        | - Running: rspec 2.rb[1:1] (n.nnnn seconds)
+        | - Failure is not order-dependent
+        |Bisect complete! Reduced necessary non-failing examples from 7 to 0 in n.nnnn seconds.
+        |
+        |The minimal reproduction command is:
+        |  rspec 2.rb[1:1]
+        EOS
```

Comment:
Not a merge blocker, but I much prefer if the lines with `|` are indented 2 extra spaces so they nested under the `expect` line.  Structurally I think of them as being "child" lines rather than "sibling" lines to the expectation line so indenting them 2 extra spaces makes that clear.  I like the `EOS` where it is.